### PR TITLE
(maint) Merge 5.5.x to 6.4.x

### PR DESCRIPTION
**Pull request by joshcooper**

### CODE REVIEW

File: acceptance/tests/language/objects_in_catalog.rb [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 tag 'audit:high',
     'audit:integration',
-    'audit:refactor'     # The use of apply on a reference system should
+    'audit:refactor',    # The use of apply on a reference system should
                          # be adequate to test puppet. Running this in
                          # context of server/agent should not be necessary.
+    'server'
```
Comment (melissa) [inline acceptance/tests/language/objects_in_catalog.rb]: This test was refactored to remove the dependency on a server host

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,ubuntu1804-64a,windows2019-64a

**Comment by melissa** [comment]
I'll merge once tests run

**Comment by joshcooper** [comment]
Sadly getting one failure due to a new test, which I'm guessing didn't exist in 5.5.x:

```
$ bundle exec rake ci:test:docker OPTIONS="--test-tag-exclude=server --test-tag-or=risk:high,audit:high"
..
Begin tests/resource/exec/should_run_command_in_cwd.rb

The Exec resource should run commands in the specified cwd
Beaker::DSL::Outcomes::FailTest: There should be one host with master defined!
...
       Total Suite Time: 404.88 seconds
      Average Test Time: 1.82 seconds
              Attempted: 223
                 Passed: 41
                 Failed: 1
                Errored: 0
                Skipped: 181
                Pending: 0
                  Total: 223
```

**Comment by joshcooper** [comment]
I'm thinking we should merge this, and then look at merging your PR against 6.4.x #7911

**Comment by joshcooper** [comment]
Oh weird, the pipeline was in-between jobs, so it was all green and looked ready to merge. But now the acceptance tests are running. Carry on!

**Pull request merged**